tools/targets/STM32_gen_PeripheralPins.py

```diff
@@ -1254,6 +1381,10 @@ def parse_BoardFile(fileName):
 
 args = parser.parse_args()
 
+if len(sys.argv)==1:
```

Comment:
Rather than manually checking no command line arguments were given, I think you could just use `parser.add_mutually_exclusive_group(required=True)` and it will display the help message if none of the required arguments are detected.